### PR TITLE
CPLAT-8357: Post-Dart2 cleanup

### PR DESCRIPTION
**Pull request by olesiathoms-wk**
## Motivation
  
We need to apply `dartfmt` fixes
## Changes
  
- `dartfmt` fixes
- remove `dart2_constant`
- update dependencies
- use base image in `Dockerfile`
- update `.travis.yaml` to run on `2.4.0`, `dev` and `stable`
## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
       - [ ] CI should be sufficient
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/w_transport/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/w_transport/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: .travis.yml [hunk 1]
```diff
@@ -10,17 +10,26 @@ cache:
...
-  - pub run dependency_validator -i build_runner,build_test,build_web_compilers,coverage,dart_style --exclude-dir=example
-  - pub run dart_dev dart1-only -- test -P integration -P travis
-  - pub run dart_dev dart2-only -- test -P integration -P travis -P dart2
+jobs:
+  include:
+    - dart: 2.4.0
+      name: "SDK: 2.4.0"
+      script:
+        - pub run dart_dev analyze
+        - pub run dart_dev dart2-only -- test -P integration -P travis -P dart2
```
Comment (todbachman-wf) [inline .travis.yml]: I think we can simplify this line to `pub run dart_dev test -P integration -P travis -P dart2`

File: .travis.yml [hunk 2]
```diff
@@ -10,17 +10,26 @@ cache:
...
+    - dart: stable
+      name: "SDK: stable"
+      script:
+        - pub run dart_dev analyze
+        - pub run dart_dev format --check
+        - pub run dependency_validator --exclude-dir=example
+        - pub run dart_dev test -P integration -P travis -P dart2
+        - pub run dart_dev dart2-only -- test -P integration -P travis -P dart2
```
Comment (todbachman-wf) [inline .travis.yml]: I think this line can be deleted. It's essentially the same as the line above it.

File: .travis.yml [hunk 3]
```diff
@@ -10,17 +10,26 @@ cache:
...
+    - dart: dev
+      name: "SDK: dev"
+      script:
+        - pub run dart_dev analyze
+        - pub run dart_dev dart2-only -- test -P integration -P travis -P dart2
```
Comment (todbachman-wf) [inline .travis.yml]: I think this line can be simplified to `pub run dart_dev test -P integration -P travis -P dart2`.

**Comment by todbachman-wf** [comment]
Should we update this repo to use `workiva_analysis_options`?

File: test/integration/global_web_socket_monitor/sockjs_port_test.dart [hunk 4]
```diff
@@ -12,13 +12,13 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-@Tags(const ['no-dart2'])
+@Tags(['no-dart2'])
```
Comment (todbachman-wf) [inline test/integration/global_web_socket_monitor/sockjs_port_test.dart]: Hmm. Can we run these tests in Dart 2 now? Also, it looks like maybe we should check to see if the `dart_test.yaml` file needs updating.

File: tool/dev.dart [hunk 5]
```diff
@@ -81,22 +81,22 @@ Future<Null> _serveExamples() {
   io.Process.start('pub', ['serve', '--port=9000'],
       workingDirectory: 'example');
 
-  return new Completer<Null>().future;
+  return Completer<Null>().future;
```
Comment (todbachman-wf) [inline tool/dev.dart]: Is the plan to update to dart_dev V3 in a followup?

Comment (olesiathoms-wk) [reply tool/dev.dart]: yeah, @evanweible-wf  has PR in progress https://github.com/Workiva/w_transport/pull/330 he started it in the summer. I asked him about it and we agreed that I'll just address `dartfmt`, `d1down` fixes and after that he'll update his PR to remove the rest of dart1.

**Comment by corwinsheahan-wf** [comment]
@Workiva/release-management-p

**Pull request merged**